server/src/main/java/org/elasticsearch/action/support/IndicesOptions.java

```diff
@@ -193,62 +482,134 @@ public boolean expandWildcardExpressions() {
      * @return Whether wildcard expressions should get expanded to hidden indices
      */
     public boolean expandWildcardsHidden() {
-        return expandWildcards.contains(WildcardStates.HIDDEN);
+        return wildcardOptions.includeHidden();
     }
 
     /**
      * @return Whether execution on closed indices is allowed.
      */
     public boolean forbidClosedIndices() {
-        return options.contains(Option.FORBID_CLOSED_INDICES);
+        return generalOptions.allowClosedIndices() == false;
     }
 
     /**
      * @return whether aliases pointing to multiple indices are allowed
      */
     public boolean allowAliasesToMultipleIndices() {
-        // true is default here, for bw comp we keep the first 16 values
-        // in the array same as before + the default value for the new flag
-        return options.contains(Option.FORBID_ALIASES_TO_MULTIPLE_INDICES) == false;
+        return generalOptions().allowAliasToMultipleIndices();
     }
 
     /**
      * @return whether aliases should be ignored (when resolving a wildcard)
      */
     public boolean ignoreAliases() {
-        return options.contains(Option.IGNORE_ALIASES);
+        return wildcardOptions.resolveAliases() == false;
     }
 
     /**
      * @return whether indices that are marked as throttled should be ignored
      */
     public boolean ignoreThrottled() {
-        return options.contains(Option.IGNORE_THROTTLED);
-    }
-
-    /**
-     * @return a copy of the {@link WildcardStates} that these indices options will expand to
-     */
-    public EnumSet<WildcardStates> expandWildcards() {
-        return EnumSet.copyOf(expandWildcards);
-    }
-
-    /**
-     * @return a copy of the {@link Option}s that these indices options will use
-     */
-    public EnumSet<Option> options() {
-        return EnumSet.copyOf(options);
+        return generalOptions().removeThrottled();
     }
 
     public void writeIndicesOptions(StreamOutput out) throws IOException {
-        out.writeEnumSet(options);
-        out.writeEnumSet(expandWildcards);
+        EnumSet<Option> backwardsCompatibleOptions = EnumSet.noneOf(Option.class);
+        if (allowNoIndices()) {
+            backwardsCompatibleOptions.add(Option.ALLOW_EMPTY_WILDCARD_EXPRESSIONS);
+        }
+        if (ignoreAliases()) {
+            backwardsCompatibleOptions.add(Option.EXCLUDE_ALIASES);
+        }
+        if (allowAliasesToMultipleIndices() == false) {
+            backwardsCompatibleOptions.add(Option.ERROR_WHEN_ALIASES_TO_MULTIPLE_INDICES);
+        }
+        if (forbidClosedIndices()) {
+            backwardsCompatibleOptions.add(Option.ERROR_WHEN_CLOSED_INDICES);
+        }
+        if (ignoreThrottled()) {
+            backwardsCompatibleOptions.add(Option.EXCLUDE_THROTTLED);
+        }
+        if (ignoreUnavailable()) {
+            backwardsCompatibleOptions.add(Option.ALLOW_UNAVAILABLE_CONCRETE_TARGETS);
+        }
+        out.writeEnumSet(backwardsCompatibleOptions);
```

Comment:
To the best of my knowledge, this is indeed using the `writeEnum` which is writing the ordinals:
```
 public <E extends Enum<E>> void writeEnum(E enumValue) throws IOException {
        assert enumValue instanceof XContentType == false : "XContentHelper#writeTo should be used for XContentType serialisation";
        writeVInt(enumValue.ordinal());
    }
```

If you think this is a bit risky, I am happy to revert the renaming, since it's so local I think it's ok to keep the previous names. I do not want any extra risks with this PR.

Reply:
I think as long as the enums _mean_ the same thing it should be fine.